Update Haskell package set to LTS 16.12 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2020-09-04 20:00 +02:00. You can watch this live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR that viewers can use to chat with me and with each other.

ingenieroariel committed:
hasura: 1.2.1 -> 1.3.1-beta.1

peti committed:
pandoc: update to version 2.10.1

Stackage wants us to stay at 2.9.x, but that version is really quite old now
and updating the new version actually simplifies our code because a couple of
overrides are no longer necessary.

peti committed:
haskell-rfc5051: update to latest version to fix pandoc-citeproc

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/ba17676480693ee7354ff25897f8cf1aefd28571.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/f23325d9cd92e534218a5281caacb8f7cbcccd33.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/f254e267909a2864552389abe416f6377916f967.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/a91f6a268b0c2e7e951113d08bb2e5953b07ab31.

peti committed:
LTS Haskell 16.12

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/c93a6c8875cce4228879f7d81e5ab96446e27320.

maralorn commented:
Question: When I understand it correctly friday is the date of branch off. Will we do this merge before branch off or will we backport it?
There are some fixes that should probably still reach 20.09.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/e85cb5656b1ef7f113518de0acd4a35aef88f65b.

rnhmjoj committed:
haskellPackages.pandoc-crossref: mark unbroken

cdepillabout committed:
Merge pull request #96956 from rnhmjoj/pandoc-crossref

haskellPackages.pandoc-crossref: mark unbroken

maralorn committed:
haskellPackages.haskell-language-server: Fix eval

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/99b594fbc83a22d6bebb43a6a63f994d472b2680.

peti commented:
> Question: When I understand it correctly friday is the date of branch off. Will we do this merge before branch off or will we backport it?

I believe that back-porting important fixes into the release branch is the way to go.

maralorn committed:
haskellPackages.monoidal-containers: Remove patch

maralorn committed:
Merge pull request #95369 from piensa/hasura-enable-update

hasura: 1.2.1 -> 1.3.1-beta.1

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/f491652f2c15d37105f94a7821159839a6661da0.

maralorn committed:
haskellPackages.ghcide: Fix build (#97052)

maralorn committed:
haskellPackages.haskell-language-server: 0.3.0 -> 0.4.0

maralorn committed:
Merge pull request #97062 from maralorn/bump-hls

haskellPackages.haskell-language-server: 0.3.0 -> 0.4.0

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-14-g12d3fb3 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/43aab36e8f475c48624aea2d96371aeb4a57be20.

peti committed:
Merge remote-tracking branch 'origin/master' into haskell-updates.

peti committed:
Merge remote-tracking branch 'origin/master' into haskell-updates.

peti committed:
hackage2nix: disable failing builds to avoid evaluation errors on Hydra

peti committed:
liquidhaskell and friends don't compile with ghc-8.8.x.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-15-g1073700 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/037ebc86aa430ea8ad5a7542132ba7a3f69ab99b.

peti committed:
haskell-neat-interpolation: update the override for the latest version

peti committed:
hackage2nix: disable failing builds to avoid evaluation errors on Hydra

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-15-g1073700 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/85155b55bca3ad2c44a2ef8633148308d1847619.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-15-g1073700 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/18ef8de8fa8811e9c250ea7e0c3dc1057151863a.

peti commented:
Next week (a.k.a. today) we'll merge https://github.com/NixOS/nixpkgs/pull/97784.
